pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -210,8 +214,11 @@ func (p *DNSProxy) checkRestored(endpointID uint64, destPort uint16, destIP stri
 	}
 
 	for i := range ipRules {
-		if _, exists := ipRules[i].IPs[destIP]; (exists || ipRules[i].IPs == nil) && ipRules[i].Re.MatchString(name) {
-			return true
+		ipRule := ipRules[i]
+		if _, exists := ipRule.IPs[destIP]; exists || ipRule.IPs == nil {
+			if ipRule.regex != nil && ipRule.regex.MatchString(name) {
```

Comment:
Is it possible for it to be `nil` or is this just coming from overly defensive coding? In the end, I think treating the `nil` as no match makes sense regardless.

Reply:
Hmm, yeah, mostly defensive since we discussed this in a previous PR in regards to the list of `matchNames`.

Reply:
> Is it possible for it to be nil

as far as i understand, the only reason for nil is if someone manually change the json file on disk to either nil or `""` (I think), so it should never happen

Reply:
Might as well keep it. We can treat it as no-match. In case someone accidentally mucks with the json file or somehow the data is corrupted, we don't want to inadvertently allow traffic, even though the chances of this happening are extremely small.